chips/earlgrey/src/pinmux.rs

```diff
@@ -0,0 +1,208 @@
+// Licensed under the Apache License, Version 2.0 or the MIT License.
+// SPDX-License-Identifier: Apache-2.0 OR MIT
+// Copyright Tock Contributors 2022.
+
+//! Mux'ing between physical pads and GPIO or other peripherals.
```

Comment:
Can we add tests for these? How do we verify that they are working

Reply:
I understand that a pinmux affects the GPIO hardware. We can still add tests that change those and can then verify them externally. It's what we do with LEDs on some platforms for example.

Otherwise we have a driver that isn't called, so it's very difficult to test and catch regressions

Reply:
How did you end up testing this?

Reply:
Short answer:  I used CW310 as DUT and Hyperdebug & opentitantool for GPIO stimulations.
But as you write we need some driver to call this code but it is't upstream on TockOS.

TL;DR:
I share your concerns about code quality but we currently don't have any decnet test framework in TockOS and OpenTitan. (that can test hardware feature like this).  We are working on some integration testing for TockOS in OpenTitan but it probably take some time. 

This PR https://github.com/tock/tock/pull/3710 will help with testing it on upstream TockOS. 
We can continue this topic in PR described above.

Reply:
Tock on OpenTitan has a few integrated unit tests which we use to test the drivers. I think the majority of drivers have at least some tests now. We have used that for a few years to catch regressions when updating the hardware. That is a good place to start for adding tests and shouldn't take too much time.

It doesn't have something to automatically check the GPIO output though. I agree that is a limitation, but at least setting the registers and ensuring they match what we expect is something. And a lot better then nothing :)